docs/source/managed-federation/deployment.mdx

```diff
@@ -117,11 +118,95 @@ The next time it starts up or polls, your router obtains an updated configuratio
 
-With managed federation, you can control which version of your graph a fleet of routers are using. In the majority of cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
+Your deployment model may require deploying a specific version of your graph, for example for blue-green deployments or canary deployments. With managed federation, you can control which version of your graph a fleet of routers are using. In the majority of cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
 
 However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage different fleets of routers running with different configurations.
 
-### Example
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning the supergraph schema of each environment to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and do analysis on the combined production traffic of all environments.
+
+### Example blue-green deployment
+
+A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic, while the other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.  
+
+As an example, follow these steps to deploy with a supergraph schema of a new release (green) environment:
+
+1. Publish all subgraphs of a release at once by using the Platform API [`publishSubgraphs`](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation#publishSubgraphs).
+
+    ```graphql
+    ## Publish multiple subgraphs together in a batch
+    ## and retrieve the associated launch, along with any downstream launches.
+    mutation PublishSubgraphsMutation(
+      $graphId: ID!
+      $graphVariant: String!
+      $revision: String!
+      $subgraphInputs: [PublishSubgraphsSubgraphInput!]!
+    ) {
+      graph(id: $graphId) {
+        publishSubgraphs( #highlight-line
+          graphVariant: $graphVariant
+          revision: $revision
+          subgraphInputs: $subgraphInputs
+        ) {
+          launch {
+            id
+            downstreamLaunches {
```

Comment:
Since this example requests information about downstream launches we should recommend users to add as an input argument `downstreamLaunchInitiation = SYNC` ([schema docs](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/enums/DownstreamLaunchInitiation#SYNC)) as part of the mutation. Otherwise these wont be created within the same request and show up as an empty list.

Reply:
Given that contracts are an enterprise only feature, do we want to have a query that doesn't request downstream launches? Maybe have a separate section for querying downstream launches which shows the example with the above mentioned input argument. 

It might be confusing that the initial recommended example will have unexpected response for non-enterprise users.

Reply:
Ah but maybe since this is under the "managed federation" section of the docs its completely fine.

Reply:
Great point, will separate this as you suggest